Tests/SwiftCliCoreTests/ChartifyTest.swift

```diff
@@ -4,6 +4,51 @@ import Nimble
 @testable import SwiftCliCore
 
 class ChartifyFinishedTest: XCTestCase {
-    // to do
+    func testRunValidatesTheInput() throws {
+        let mock = MockInputValidator()
+        Chartify().run(userInput: "k1", inputValidator: mock, chartConstructor: ChartConstructor())
+        let result = mock.wasValidateCalled
+        let expectedResult = true
+    expect(result).to(equal(expectedResult))
+}
+
+    func testRunTransformsTheInputIntoAnArrayIfInputIsValid() throws {
+        let mock = MockInputValidator()
+        Chartify().run(userInput: "k1", inputValidator: mock, chartConstructor: ChartConstructor())
+        let result = mock.wasArrayMakerCalled
+        let expectedResult = true
+        expect(result).to(equal(expectedResult))
+    }
+
+    func testRunMakesTheChartIfInputIsValid() throws {
+        let mock = MockChartConstructor()
+        Chartify().run(userInput: "k1", inputValidator: InputValidator(), chartConstructor: mock)
+        let result = mock.wasMakeChartCalled
+        let expectedResult = true
+        expect(result).to(equal(expectedResult))
+    }
+
```

Comment:
Ideally, you would have a test case here that the error message was printed if the input is invalid. I'm not going to insist on it, but think about what changes you might make to your code in order to test that.

Reply:
I think if I change Validator to throw an error that is then printed, that would go a way towards making things testable.